src/test/ui/traits/reservation-impls/reservation-impl-ok.rs

```diff
@@ -0,0 +1,28 @@
+// run-pass
+
+// rpass test for reservation impls. Not 100% required because `From` uses them,
+// but still.
+
+#![feature(rustc_attrs)]
+
+use std::mem;
+
+trait MyTrait<S> {
+    fn foo(&self, s: S) -> usize;
+}
+
+#[rustc_reservation_impl]
+impl<T> MyTrait<u64> for T {
+    fn foo(&self, _x: u64) -> usize { 0 }
+}
+
+// reservation impls don't create coherence conflicts, even with
+// non-chain overlap.
```

Comment:
So I guess this implies that someone could implement `impl From<!> for MyType` -- well, I guess that something we wanted, IIRC.

Reply:
I don't see a reason not to just have a blanket impl:

```
impl<T> From<!> for T {
    fn from(_: !) -> T {
        unimplemented!()
    }
}
```

This is sound because it can never be called.

Also, I need this for something like [this](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=08baed31308bc88cbcadec1951e9778b) :)

Reply:
Yea this is intentional.